src/js/containers/generateFiles/GenerateFilesContainer.jsx

```diff
@@ -338,9 +339,11 @@ class GenerateFilesContainer extends React.Component {
         });
     }
 
-    toggleAgencyType() {
+    toggleAgencyType(type) {
+        const newType = _.assign({}, this.state[type]);
+        newType.fundingAgency = !newType.fundingAgency;
         this.setState({
-            fundingAgency: !this.state.fundingAgency
+            [type]: newType
         });
```

Comment:
Some feedback on implementation:

`setState` comes with the `prevState` parameter you can use to accomplish the same thing without having to create a new const:

```javascript
this.setState((prevState) => ({
  [type]: {
     ...prevState[type],
    isFundingAgency: !prevState[type].isFundingAgency
  }
})
```

Reply:
Thank you.  I was told by @ebdabbs not to update properties in `setState`.